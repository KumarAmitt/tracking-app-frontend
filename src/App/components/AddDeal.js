import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Select from 'react-select';
import { Redirect } from 'react-router-dom';
import AppBar from './AppBar';
import { getProducts, loadProducts } from '../../store/slicers/product';
import { createDeal } from '../../store/slicers/deal';
import { getSessionInfo, loadSession } from '../../store/slicers/user_session';
import './style/AddDeal.css';

const AddDeal = () => {
  const appBarTitle = 'Add Deal';
  const dispatch = useDispatch();
  const products = useSelector(getProducts);
  const sessionInfo = useSelector(getSessionInfo);
  const [dealInfo, setDealInfo] = useState({ productId: 0, applicationId: '', premium: '' });

  useEffect(() => {
    dispatch(loadProducts());
    dispatch(loadSession());
  }, []);

  const handleChange = (e) => {
    setDealInfo({ ...dealInfo, [e.target.name]: e.target.value });
  };

  const handleSelect = (e) => {
    setDealInfo({ ...dealInfo, productId: e });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (dealInfo.productId === 0) {
      const el = document.getElementById('select-placeholder');
      el.innerHTML = 'Product must be selected';
      el.style.color = 'red';
    } else {
      // console.log(dealInfo);
      dispatch(createDeal({
        product_id: dealInfo.productId,
        application_id: dealInfo.applicationId,
        premium: dealInfo.premium,
      }));
      setDealInfo({
        ...dealInfo, productId: 0, applicationId: '', premium: '',
      });
    }
  };

  if (!sessionInfo.logged_in) {
    return <Redirect to="/" />;
  }

  // console.log(products);
  return (
    <>
      <AppBar title={appBarTitle} link="/track" />
      <div className="add-deal">
        <form onSubmit={handleSubmit}>
          <Select
            className="react-select"
            classNamePrefix="react-select"
            options={products.map((p) => ({ label: p.product_name, value: p.id }))}
            onChange={(e) => handleSelect(e.value)}
            placeholder={<div id="select-placeholder">Select a product</div>}
          />
          <input type="text" name="applicationId" placeholder="Application ID" value={dealInfo.applicationId} onChange={handleChange} />
          <input type="number" step="0.01" name="premium" placeholder="Premium" value={dealInfo.premium} onChange={handleChange} required />
          <button type="submit">Submit</button>
        </form>
      </div>
    </>
  );
};

export default AddDeal;
