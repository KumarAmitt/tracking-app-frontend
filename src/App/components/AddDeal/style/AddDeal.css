.add-deal {
  margin-top: 3em;
  height: calc(100vh - 7em);
}

.react-select .react-select__control {
  border: none;
}

.add-deal form > * + * {
  margin: 1em auto;
  width: 90%;
}

.red-text {
  color: red;
}
