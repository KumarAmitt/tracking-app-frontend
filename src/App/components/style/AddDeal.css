.react-select .react-select__control {
  border: none;
}

.add-deal form > * + * {
   margin: 1em auto;
    width: 90%;
}
