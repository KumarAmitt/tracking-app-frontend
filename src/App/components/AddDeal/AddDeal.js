import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Select from 'react-select';
import { Redirect } from 'react-router-dom';
import AppBar from '../AppBar/AppBar';
import { getProducts, loadProducts } from '../../../store/slicers/product';
import { createDeal } from '../../../store/slicers/deal';
import './style/AddDeal.css';

const AddDeal = () => {
  const appBarTitle = 'Add Deal';
  const dispatch = useDispatch();
  const products = useSelector(getProducts);

  const [productId, setProductId] = useState(0);
  const [applicationId, setApplicationId] = useState('');
  const [premium, setPremium] = useState('');

  const changeApplicationId = (e) => {
    setApplicationId(e.target.value);
  };

  const changePremium = (e) => {
    setPremium(e.target.value);
  };

  const changeProductId = (e) => {
    setProductId(e);
  };

  const resetForm = () => {
    setProductId(0);
    setApplicationId('');
    setPremium('');
  };

  useEffect(() => {
    dispatch(loadProducts());
  }, []);

  const handleSubmit = (e) => {
    e.preventDefault();
    if (productId === 0) {
      const el = document.getElementById('select-placeholder');
      el.innerHTML = 'Product must be selected';
      el.style.color = 'red';
    } else {
      dispatch(createDeal({
        product_id: productId,
        application_id: applicationId,
        premium,
      }));
      resetForm();
    }
  };

  // if (!sessionInfo.logged_in) {
  //   return <Redirect to="/" />;
  // }

  return (
    <>
      <AppBar title={appBarTitle} link="/track" />
      <div className="add-deal">
        <form onSubmit={handleSubmit}>
          <Select
            className="react-select"
            classNamePrefix="react-select"
            options={products.map((p) => ({ label: p.product_name, value: p.id }))}
            onChange={(e) => changeProductId(e.value)}
            placeholder={<div id="select-placeholder">Select a product</div>}
          />
          <input type="text" placeholder="Application ID" value={applicationId} onChange={changeApplicationId} />
          <input type="number" step="0.01" placeholder="Premium" value={premium} onChange={changePremium} required />
          <button type="submit" data-testid="submit">Submit</button>
        </form>
      </div>
    </>
  );
};

export default AddDeal;
