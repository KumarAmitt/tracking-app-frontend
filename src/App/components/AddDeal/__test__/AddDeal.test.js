import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import AddDeal from '../AddDeal';
import configureAppStore from '../../../../store/configureStore';

const store = configureAppStore();

describe('AddDeal', () => {
  test('should match the snapshot of AddDeal component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <AddDeal />
        </BrowserRouter>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
