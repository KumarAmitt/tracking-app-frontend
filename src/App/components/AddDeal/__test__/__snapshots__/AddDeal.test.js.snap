// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDeal should match the snapshot of AddDeal component 1`] = `null`;
